### PR TITLE
Getting close to synced.

### PR DESCRIPTION
**Pull request by dabercro**
for comments

### CODE REVIEW

File: slimmer/slimmer.cpp [hunk 1]
```diff
@@ -122,10 +122,20 @@ int parsed_main(int argc, char** argv) {
...
       //// PHOTONS ////
 
+      std::vector<std::pair<float, float>> em_directions; // Pairs of eta, phi for preselected leptons for cleaning
+
       for (auto& pho : event.photons) {
         output.n_pho_loose += pho.loose;
         output.n_pho_medium += pho.medium;
         output.n_pho_tight += pho.tight;
+        // if (pho.medium && pho.csafeVeto && pho.pixelVeto && pho.pt() > 25 && std::abs(pho.eta()) < 2.5)
+        if (pho.medium && pho.pt() > 175 && std::abs(pho.eta()) < 2.5 && pho.loose && pho.csafeVeto) {
```
Comment (dabercro) [inline slimmer/slimmer.cpp:132]: @DylanHsu I don't think we should require this photon pt to be so high before cleaning. (Though I wasn't cleaning at all before the sync)

File: slimmer/slimmer.cpp [hunk 2]
```diff
@@ -191,54 +199,62 @@ int parsed_main(int argc, char** argv) {
...
-                        lep.ecalIso/lep.pt() < 0.45 and lep.hcalIso/lep.pt() < 0.28)
-                       );
+                    return true;
+                    // return lep.smearedPt > 15.0 and lep.hOverE < 0.09 and reliso < 0.18 and
+                    //   ((abseta < 1.4442 and lep.sieie < 0.012 and
+                    //     lep.ecalIso/lep.smearedPt < 0.4 and lep.hcalIso/lep.smearedPt < 0.25 and
+                    //     std::abs(lep.dEtaInSeed) < 0.0095 and std::abs(lep.dPhiIn) < 0.065) or
+                    //    (abseta > 1.5660 and lep.sieie < 0.033 and
+                    //     lep.ecalIso/lep.smearedPt < 0.45 and lep.hcalIso/lep.smearedPt < 0.28)
+                    //    );
```
Comment (dabercro) [inline slimmer/slimmer.cpp:253]: @DylanHsu I think we should also include at least these eta cuts for tight lepton counting, though we also want to veto on stuff that goes in there. I'm not sure what is best.

File: slimmer/slimmer.cpp [hunk 3]
```diff
@@ -350,15 +366,15 @@ int parsed_main(int argc, char** argv) {
...
+        if (overlap_em(jet) || jet.pt() < 20.0)
           continue;
 
         // Count jets (including forward)
         auto abseta = std::abs(jet.eta());
         output.set_countjets(hbbfile::countjets::fake, jet, abseta);
 
-        if (not jet.loose)
-          continue;
+        // if (not jet.loose)
```
Comment (dabercro) [inline slimmer/slimmer.cpp:376]: @DylanHsu I think we should really require jets to be loose, because they look super fake otherwise

Comment (DylanHsu) [reply slimmer/slimmer.cpp:376]: ok, we can probably add the `analysis->hbb` case here
https://github.com/sidnarayanan/PandaAnalysis/blob/master/Flat/src/ModulesJets.cc#L69

**Comment by dabercro** [comment]
@maierbenedikt you may want to comment on these cut choices too.

File: slimmer/slimmer.cpp (hunk 2 above)
Comment (DylanHsu) [reply slimmer/slimmer.cpp:253]: from what it looks like in the analysis note, the electron preselection includes the ECAL crack, but the tight object definition does not include the ECAL crack

Comment (DylanHsu) [reply slimmer/slimmer.cpp:253]: we'll add the cuts you commented out to the MVAWP80 and MVAWP90 bit definition, i.e. here:
https://github.com/sidnarayanan/PandaAnalysis/blob/master/Flat/src/ModulesLepPho.cc#L34-L35

File: slimmer/slimmer.cpp (hunk 1 above)
Comment (DylanHsu) [reply slimmer/slimmer.cpp:132]: From what we know, the nominal analysis is not doing photon-jet cleaning at all. So we should either not do it at all or have the photon definition for cleaning be very tight. Have you found anything indicating that the other group does photon cleaning and if so what cuts do they use?

File: slimmer/slimmer.cpp [hunk 4]
```diff
@@ -160,14 +168,14 @@ int parsed_main(int argc, char** argv) {
                       << reliso << "]" << std::endl;
           }
           em_directions.emplace_back(lep.eta(), lep.phi());
+          output.n_lep_loose++;
           if (is_loose()) {
-            lepvec += lep.p4();
-            output.n_lep_loose++;
```
Comment (dabercro) [inline slimmer/slimmer.cpp:165]: @DylanHsu Here, I moved from counting loose leptons that passed the loose cut to counting leptons that pass the preselection. I got the impression from the AN that the preselection is used for cleaning, but the loose counting is done with the MVA/muon working points. I could be wrong about that though.

Comment (DylanHsu) [reply slimmer/slimmer.cpp:165]: for what it's worth I think that is the right interpretation

**Pull request merged**